### PR TITLE
Migrate wiki page: Debugging

### PR DESCRIPTION
**Pull request by seanmadsen**
An import of the wiki page on debugging

### CODE REVIEW

File: docs/dev-tools/debugging.md [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+
+```bash
+sudo yum install php-pecl* php-devel php-pear
+sudo pecl install Xdebug
+```
+
+Mac OS X
+
+```bash
+sudo port install php5-xdebug
```
Comment (totten) [inline docs/dev-tools/debugging.md]: OS X users have many different (but never-the-less common) ways of running PHP (including but not limited to Apple's built-in PHP,  `brew`, MAMP, XAMPP, AMPP, Vagrant, Bitnami, and MacPorts)... but this instruction only applies to MacPorts. Don't think I've encountered OS X users doing Civi on MacPorts for a while.

It kinda sucks, but the best advice I've come up with for OS X users is to identify your specific runtime and search Google, eg `mamp xdebug` or `macports xdebug`.

Comment (seanmadsen) [inline docs/dev-tools/debugging.md]: Thanks for pointing this out @totten. I added some of what your wrote in the commit below.

**Comment by seanmadsen** [comment]
@totten do you think we can merge this now, given then changes I made to address your feedback?

**Comment by totten** [comment]
👍  Yeah, let's do it. 

I may propose a few more edits.

**Pull request merged**